File: python/news.py
# news.py
import os
from API import BraveAPI
from utils.main import dump_schema

# Create an instance of the BraveAPI class
brave = BraveAPI({
    'subscription_token': os.environ['BRAVE_KEY']
})

# Setting up parameters
params = {
    'q': "San Francisco Hackathon Shack15",
    'text_decorations': False,
    'freshness': "py"
}

# Performing the news search
results = brave.news(params)

# Dumping and printing the schema of the results
print(dump_schema(results))

File: js/news.js
// news.js
const { dumpSchema } = require("./utils/index.js");
const BraveAPI = require("./API.js");

// Create a new instance of the BraveAPI class
const brave = new BraveAPI({
    subscription_token: process.env.BRAVE_KEY
});

// Setup the search parameters
const params = {
    q: "San Francisco Hackathon Shack15",
    text_decorations: false,
    freshness: "py",
};

// Perform the search
brave.news(params).then(results => {
    // Log the results
    console.log(dumpSchema(results));
});
